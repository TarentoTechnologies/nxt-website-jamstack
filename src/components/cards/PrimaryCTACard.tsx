import { Link } from "gatsby";
import React from "react";

import { black75 } from "../../styles/style-guide/ColorGuide.module.css";
import {
  ctaDescription,
  h1One,
} from "../../styles/style-guide/Typography.module.css";
import { PrimaryCTA } from "../buttons/PrimaryCTA";

interface PrimaryCTACardProps {
  title: string;
  description?: string;
  ctaText: string;
  ctaLink: string;
  isCTAMail?: boolean;
  isCTAExternal?: boolean;
}

export const PrimaryCTACard = ({
  title,
  description,
  ctaText,
  ctaLink,
  isCTAMail,
  isCTAExternal,
}: PrimaryCTACardProps) => {
  return (
    <div className={`container-fluid`} data-testid="primary-cta-card">
      <div className="">
        <h1 className={`${h1One}`} data-aos="fade-up">
          {title}
        </h1>
        <p
          className={`col-sm-10 col-md-10 col-lg-10 col-xl-6 mt-4 mb-5 ${black75} ${ctaDescription}`}
          data-aos="fade-up"
        >
          {description}
        </p>
        <span data-aos="fade-up">
          <PrimaryCTA
            label={ctaText}
            btnLink={ctaLink}
            isMail={isCTAMail}
            isExternal={isCTAExternal}
          />
        </span>
      </div>
    </div>
  );
};
