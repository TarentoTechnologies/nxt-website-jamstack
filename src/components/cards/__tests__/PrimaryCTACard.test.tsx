import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { PrimaryCTACard } from "../PrimaryCTACard";

describe("Test if the PrimaryCTACard component is working as expected", () => {
  it("<PrimaryCTACard /> matches snapshot", () => {
    const component = render(
      <PrimaryCTACard
        title={"Are you interested ?"}
        description="Do these accelerators look interesting to you? Get in touch with us to know how to buy any of them. Click on the button below to send us a mail."
        ctaText={"Contact us"}
        ctaLink={"/"}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<PrimaryCTACard /> renders without exploding", () => {
    const component = render(
      <PrimaryCTACard
        title={"Are you interested ?"}
        description="Do these accelerators look interesting to you? Get in touch with us to know how to buy any of them. Click on the button below to send us a mail."
        ctaText={"Contact us"}
        ctaLink={"/"}
      />
    );
    expect(component.getByTestId("primary-cta-card")).toBeInTheDocument();
  });
});
