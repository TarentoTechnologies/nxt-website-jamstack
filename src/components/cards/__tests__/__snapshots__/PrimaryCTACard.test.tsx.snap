// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the PrimaryCTACard component is working as expected <PrimaryCTACard /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid"
    data-testid="primary-cta-card"
  >
    <div
      class=""
    >
      <h1
        class="h1One"
        data-aos="fade-up"
      >
        Are you interested ?
      </h1>
      <p
        class="col-sm-10 col-md-10 col-lg-10 col-xl-6 mt-4 mb-5 black75 ctaDescription"
        data-aos="fade-up"
      >
        Do these accelerators look interesting to you? Get in touch with us to know how to buy any of them. Click on the button below to send us a mail.
      </p>
      <span
        data-aos="fade-up"
      >
        <a
          data-testid="primary-cta"
          href="/"
        >
          <div
            class="text-center d-flex align-items-center justify-content-center 
      secondaryBtn  undefined"
            data-aos="fade-up"
          >
            <label
              class="labelStyles"
            >
              Contact us
            </label>
          </div>
        </a>
      </span>
    </div>
  </div>
</div>
`;
